### ISSUE TITLE
unexpected output from "qgis:intersection"

### ISSUE THREAD

**Issue by tholdaway** [jannes-m/RQGIS#74]
I am trying to intersect some SpatialPolygonsDataFrames with each other using the qgis:intersection algorithm. However, I am getting some unexpected results. Namely, all calls to run_qgis with alg = "qgis:intersection" after the first return the same thing.

A minimal example:

```
# some example SpatialPolygonsDataFrames
library(sp)
crds <- cbind(x=c(0, 0, 400, 400, 0), y=c(0, 400, 400, 0, 0))
Pl <- Polygon(crds)
ID <- "1"
Pls <- Polygons(list(Pl), ID=ID)
SPls <- SpatialPolygons(list(Pls))
df <- data.frame(value=1, row.names=ID)
SPDF1 <- SpatialPolygonsDataFrame(SPls, df)
crds <- cbind(x=c(0, 0, 400, 400, 0), y=c(0, 400, 400, 0, 0)) - 200
Pl <- Polygon(crds)
ID <- "2"
Pls <- Polygons(list(Pl), ID=ID)
SPls <- SpatialPolygons(list(Pls))
df <- data.frame(value=1, row.names=ID)
SPDF2 <- SpatialPolygonsDataFrame(SPls, df)
crds <- cbind(x=c(0, 0, 400, 400, 0), y=c(0, 400, 400, 0, 0)) + 200
Pl <- Polygon(crds)
ID <- "3"
Pls <- Polygons(list(Pl), ID=ID)
SPls <- SpatialPolygons(list(Pls))
df <- data.frame(value=1, row.names=ID)
SPDF3 <- SpatialPolygonsDataFrame(SPls, df)


library(RQGIS)
RQGIS::set_env("/Applications/QGIS.app")
test1 <- RQGIS::run_qgis("qgis:intersection", INPUT = SPDF1, INPUT2 = SPDF2, OUTPUT = file.path(tempdir(), "test1.shp"), load_output = TRUE)

test2 <- RQGIS::run_qgis("qgis:intersection", INPUT = SPDF1, INPUT2 = SPDF3, OUTPUT = file.path(tempdir(), "test2.shp"), load_output = TRUE)

identical(test1, test2)
# TRUE
```

But in reality SPDF1 intersect with SPDF2 is not equal to SPDF1 intersect with SPDF3.

My session info:

> R version 3.4.0 (2017-04-21)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

>Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

>locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

>attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

>other attached packages:
[1] RQGIS_1.0.1.9000 sp_1.2-4        

>loaded via a namespace (and not attached):
 [1] Rcpp_0.12.11          lattice_0.20-35       sf_0.4-3              grid_3.4.0            R6_2.2.1              jsonlite_1.5         
 [7] DBI_0.6-1             magrittr_1.5          units_0.4-4           stringi_1.1.5         rlang_0.1.1           raster_2.5-8         
[13] reticulate_0.9.0.9001 rgdal_1.2-7           tools_3.4.0           stringr_1.2.0         udunits2_0.13         readr_1.1.1          
[19] hms_0.3               parallel_3.4.0        compiler_3.4.0        tibble_1.3.3  

and I am using QGIS 2.18.7 from Kyngchaos.

**Comment by jannes-m** [comment]
I am sorry for the late reply but we were on the UseR-conference last week. 
Thank you for your reproducible example. To be honest, it leaves me a bit puzzled. Under Windows everything works as expected. Under Linux and Mac it doesn't... 
@pat-s Any thoughts on this? If not, I guess we have to file an issue on Stack Overflow or the QGIS forum.

In the meantime, use GRASS:

    plot(SPDF1)
    plot(SPDF2, add = TRUE)
    plot(SPDF3, add = TRUE)
    a <- run_qgis("grass7:v.overlay", ainput = SPDF1, binput = SPDF2, 
                  output = "test1.shp", load_output = TRUE)
    plot(a, add = TRUE, col = "blue")
    b <- run_qgis("grass7:v.overlay", ainput = SPDF1, binput = SPDF3, 
                  output = "test2.shp", load_output = TRUE)
    plot(b, add = TRUE, col = "red")

**Comment by pat-s** [comment]
It seems that this is a QGIS issue rather than a `RQGIS` one. In the end, we only use the algs from QGIS. I would first check if this issue persists within QGIS and if it does, I would open a ticket in the QGIS issue hub and (if there is no response) on SO. However, if its a clear issue/bug within QGIS, SO will not help that much. 

Great that there exists a GRASS workaround - its good to have alternatives from other providers if one is buggy. That's a good example which demonstrates the strength of QGIS/RQGIS :)

**Comment by tholdaway** [comment]
Thanks. I'll look into reproducing with just QGIS.

**Comment by pat-s** [comment]
Closing due to missing activity. Feel free to re-open if needed.

**Comment by jannes-m** [comment]
The issue still persists. We should file a ticket. But before doing so, let us collect all issues we have encountered.

**Comment by jannes-m** [comment]
#tholdaway Note that we found a solution for the problem you identified. Have a look at the recent discussion in #82. For now, you have to use the unlock-branch but I will work on integrating it into the master branch.

**Comment by jannes-m** [comment]
Finally resolved through 494e666 thanks to input by @lbusett (#82)
